src/Groups/directproducts.jl

```diff
@@ -372,10 +382,12 @@ Group homomorphism
   to aut( <pc group of size 8 with 3 generators> )
 
 julia> G = semidirect_product(Q,f,C)
-SemidirectProduct( <pc group of size 8 with 3 generators> , <pc group of size 2 with 1 generator> )
+Semidirect product of
+  pc group of order 8
+  pc group of order 2
 
 julia> derived_subgroup(G)
-(Group([ f4 ]), Hom: group([ f4 ]) -> semidirectProduct( <pc group of size 8 with 3 generators> , <pc group of size 2 with 1 generator> ))
+(Subgroup of Q : C, Hom: subgroup of semidirect product of groups -> semidirect product of groups)
```

Comment:
```suggestion
(Subgroup of Q : C, Hom: subgroup of Q : C -> semidirect product of groups)
```
this would be nicer, but I don't know what needs to be changed in the show function for this to happen

Reply:
yeah I agree. Hmm, I think I may even have an issue about this somewhere that it is a bit weird as to when we use `@show_name` and when not...